test/integration_test.py

```diff
@@ -387,16 +388,23 @@ def _list_managed_access_bundles(self,
         # prefix as possible.
         for source in self.managed_access_sources_by_catalog[catalog]:
             assert str(source.spec) in sources
-            bundle_fqid = self.random.choice(
-                self.azul_client.list_bundles(catalog, source, prefix='')
-            )
-            # FIXME: We shouldn't need to include the common prefix
-            #        https://github.com/DataBiosphere/azul/issues/3579
-            common = source.spec.prefix.common
-            prefix = bundle_fqid.uuid[len(common):8]
-            assert prefix != '', prefix
-            new_fqids = self.azul_client.list_bundles(catalog, source, prefix)
-            yield source, prefix, new_fqids
+            bundles = self.azul_client.list_bundles(catalog, source, prefix='')
+            bundles.sort()
+            while True:
+                bundle_fqid = self.random.choice(bundles)
+                if config.is_anvil_enabled(catalog):
+                    assert isinstance(bundle_fqid, TDRAnvilBundleFQID)
+                    # DUOS bundles are too sparse to fulfill the managed access tests
+                    if bundle_fqid.entity_type is BundleEntityType.duos:
+                        continue
```

Comment:
This would enter an infinite loop if all elements of `bundles` were DUOS. Unlikely to occur, but still.

I prefer an approach that does not make use of bailouts. Like filtering all DUOS bundles out, and asserting that the result is not empty, before picking a random element.

Reply:
I think an assertion would be redundant since `Random.choice` already raises an exception on an empty iterable:

```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/nadove/.pyenv/versions/3.11.7/lib/python3.11/random.py", line 373, in choice
    raise IndexError('Cannot choose from an empty sequence')
IndexError: Cannot choose from an empty sequence
```